api/models/usersDirectMessageThreads.js

```diff
@@ -103,6 +104,30 @@ const updateDirectMessageThreadNotificationStatusForUser = (
 ===========================================================
 */
 
+// TODO: Add method to update if the thread is archived or not.
+// TODO: Add method to get if the thread is archived or not.
+const updateIsDirectMessageThreadArchived = (
+  userDirectMessageThreadId: string,
+  isArchived: boolean
+): Promise<Object> => {
+  return db
+    .table('usersDirectMessageThreads')
+    .get(userDirectMessageThreadId)
+    .update({ isArchived })
+    .run();
+};
+
+const getDirectMessageThread = (
+  threadId: string,
+  userId: string
+): Promise<Object> => {
+  return db
+    .table('usersDirectMessageThreads')
+    .filter({ threadId })
+    .filter({ userId })
```

Comment:
RethinkDB-specific performance note: Using `.filter` is reeeaaaallly slow. It iterates through the entire set before the `.filter`, so in this case this will check every single record in the table. Instead, you can use `.getAll` with an index, which is achieves the same thing much much faster!

We already have indexes for both `userId` and `threadId` on the `usersDirectMessageThreads` table, so you can use either in combination with `.getAll` and it'll be much much much faster:

```JS
return db
  .table('usersDirectMessageThreads')
  // First, use the threadId index to quickly get all records for this thread
  .getAll(threadId, { index: 'threadId' })
  // ...and only then filter by userId to get the specific record we want
  // Here, filtering is fine because here it's only 2-10 records we're filtering through, not the entire table
  .filter({ userId })
  .run();
```

We could also do it the other way around (first use `getAll` with the `userId` index, then filter by `threadId`), but my intuition is that people usually have more DM threads total than participants in DM threads, so this way _should_ be faster.